lib/Auth/Process/Mfa.php

```diff
@@ -303,29 +280,28 @@ public static function validateMfaSubmission(
         
         try {
             $idBrokerClient = self::getIdBrokerClient($state['idBrokerConfig']);
-            $authenticatedUser = $idBrokerClient->mfaVerify(
+            $validMfa = $idBrokerClient->mfaVerify(
                 $mfaId,
                 $employeeId,
                 $mfaSubmission
             );
-            if ($authenticatedUser === null) {
+            if ( ! $validMfa) {
                 return 'Incorrect 2-step verification code.';
             }
         } catch (\Throwable $t) {
             $logger = new Psr3SamlLogger();
             $logger->critical($t->getCode() . ': ' . $t->getMessage());
             return 'Something went wrong while we were trying to do the '
-                 . '2-step verification.';
+                 . '2-step verification.' . $t->getMessage();
         }
-        
-        /* Save the attributes we received from the login-function in the $state-array. */
-        assert('is_array($attributes)');
-        $state['Attributes'] = array_replace_recursive(
-            $state['Attributes'],
-            $authenticatedUser
-        );
-        
+
+        // Set remember me cookies if requested
+        if ($rememberMe) {
+            self::setRememberMeCookies($state['employeeId'], $state['mfaOptions']);
+        }
+
         // The following function call will never return.
```

Comment:
Low priority, but all of these comments about the following function call never returning applies to the `resumeProcessing()` not `unset()`, which is no longer the impression given.